src/multiom/api/api_encode_c.cc

```diff
@@ -54,27 +54,31 @@ int multio_grib2_encoder_encode64(void* multio_grib2, void* mars_dict, void* par
         free(scaleFactorCStr);
     }
 
-    std::vector<double> scaledValues;
-    if (scaleFactor && (*scaleFactor != 1.0)) {
-        scaledValues.reserve(data_len);
-        std::transform(data, data + data_len, std::back_inserter(scaledValues),
-                       [&scaleFactor](const double& v) { return v * (*scaleFactor); });
-        data = scaledValues.data();
-    }
-
     char* missingValuesCount = "0";
     ret = multio_grib2_dict_get(par_dict, "number-of-missing-values", &missingValuesCount);
     long bitmapPresent = static_cast<long>(static_cast<bool>(std::stol(missingValuesCount)));
```

Comment:
Not part of the PR but: 
`static_cast<long>(std::stol(missingValuesCount) > 0);`
Should also work and could be more readable.